File: Java Based Archives/ViewModelDI/app/src/main/java/com/android/viewmodel/MainActivityViewModelFactory.java
package com.android.viewmodel;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MainActivityViewModelFactory implements ViewModelProvider.Factory {
    private int initialValue;

    public MainActivityViewModelFactory(int initialValue) {
        this.initialValue = initialValue;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        if (modelClass.isAssignableFrom(MainActivityViewModel.class)) {
            return (T) new MainActivityViewModel(initialValue);
        }
        throw new IllegalArgumentException("Unknown ViewModel class");
    }
}

File: JetPackCompose/StateFlow/app/src/main/java/com/android/stateflow/MainActivityViewModelFactory.kt
package com.example.viewmodeldemo2

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.android.stateflow.MainActivityViewModel

class MainActivityViewModelFactory(private val startingTotal : Int) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T = MainActivityViewModel(startingTotal) as T
}